src/Types/TypeRegistry.php

```diff
@@ -26,6 +26,37 @@ public function __construct(array $instances = [])
         $this->instances = $instances;
     }
 
+    public static function builtIn(): self
```

Comment:
IMO, the introduction of a new public method w/o understanding the full upgrade path isn't the right step. It doesn't solve any immediate problem but potentially increases the BC break in the future.

Purely from the API design standpoint, the built-in type mapping should be the default w/o needing any API to request it. For instance, the set of the defaults could be implemented in the `Configuration` class, not here.

Reply:
You're right, it's wrong to be able not to get the built-in type mapping… but that's already the case if you instantiate the `TypeRegistry` manually. Maybe that class should have been marked as internal?

If we move the set of defaults to the `Configuration`, then we should have a `Configuration::getTypeRegistry()` method I suppose?

Reply:
> it's wrong to be able not to get the built-in type mapping…

I wouldn't say it's wrong. I'd say it shouldn't require an extra effort to get the default.

> If we move the set of defaults to the `Configuration`, then we should have a `Configuration::getTypeRegistry()` method I suppose?

That would make sense. But again, I would not like to introduce a new public API without understanding the full idea first.

Reply:
Okay, let's close this 👍